src/widgets/editor.rs

```diff
@@ -17,7 +17,7 @@ use widgets::{View, ViewClient};
 pub struct Editor {
     pub pending_open_requests: Vec<ClientResult<(ViewId, View)>>,
     pub delayed_events: Vec<CoreEvent>,
-    pub views: HashMap<ViewId, View>,
+    pub views: IndexMap<ViewId, View>,
```

Comment:
HashMap's don't ensure the order of the entries, so in order to keep a list of buffers capable of left and right motion we would have to keep a list containing all view id's
from the hasmap and keeping it synced with `views` would be tedious.

indexmap does ensure the elements are kept in insertion order, so it's just as simple as referencing the entries with an index.

Reply:
HashMaps aren't ordered, so the values don't always remain in insertion order. IndexMap does so we can simply increment the index of the current buffer instead of having to keep track of what order the views are in.

Reply:
@little-dude I've made all the changes you requested on a local repo, I'm only waiting to resolve this  to push the final changes